File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        left = 0
        right = n-1
        index = n-1
        result = [0 for i in range(n)]
        
        while(index>=0):
            if abs(nums[left])>=abs(nums[right]):
                result[index]=nums[left]*nums[left]
                index-=1
                left+=1
            else:
                result[index]=nums[right]*nums[right]
                index-=1
                right-=1
                
        return result        

File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int l=0,r=nums.size()-1;
        vector<int>v(nums.size());
        for(int k=nums.size()-1;k>=0;k--){
            if(abs(nums[l])>abs(nums[r]))
                v[k]=nums[l]*nums[l++];
            else
                v[k]=nums[r]*nums[r--];}
        return v;
    }
};